### PR TITLE
reworks data handling for lock and increase lock amount

### PR DESCRIPTION
**Pull request by Emmanueldmlr**
# reworks data handling for lock and increases lock amount

fixes https://github.com/EveripediaNetwork/issues/issues/872

### CODE REVIEW

File: src/hooks/useLock.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export const useLock = () => {
   }
 
   const lockIQ = async (amount: number, lockPeriod: number) => {
-    const parsedAmount = ethers.utils.parseEther(amount.toString())
+    const parsedAmount = ethers.utils.parseEther(`${amount}`)
```
Comment (kesar) [inline src/hooks/useLock.ts:53]:
```suggestion
    const parsedAmount = ethers.utils.parseEther(amount)
```

File: src/hooks/useLock.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ export const useLock = () => {
   }
 
   const increaseLockAmount = async (amount: number) => {
-    const parsedAmount = ethers.utils.parseEther(amount.toString())
+    const parsedAmount = ethers.utils.parseEther(`${amount}`)
```
Comment (kesar) [inline src/hooks/useLock.ts:69]:
```suggestion
    const parsedAmount = ethers.utils.parseEther(amount)
```

File: src/hooks/useLock.ts (hunk 1 above)
Comment (Emmanueldmlr) [reply src/hooks/useLock.ts:53]: @kesar parseEther takes a string as its parameter.

File: src/components/lock/StakeIQ.tsx [hunk 3]
```diff
@@ -96,12 +103,11 @@ const StakeIQ = ({ exchangeRate }: { exchangeRate: number }) => {
     }
   }, [data, toast, trxHash, checkPoint])
 
-  const updateIqToBeLocked = (value: string | number) => {
+  const updateIqToBeLocked = (value: Result | number | string) => {
```
Comment (kesar) [inline src/components/lock/StakeIQ.tsx:106]: a function that accepts strings or result or number its a code smell that code is not very solid and might lead to unexpected results.

why there are multiple ways of update IQ w different formats? There should be only one to avoid having a bug sending a string vs sending a number vs sending a bigInt

File: src/hooks/useErc20.ts [hunk 4]
```diff
@@ -28,8 +28,7 @@ export const useErc20 = () => {
 
   const getUserBalance = () => {
     if (erc20Balance) {
-      const result = formatContractResult(erc20Balance)
-      return result
+      return erc20Balance
     }
     return 0
```
Comment (kesar) [inline src/hooks/useErc20.ts:33]: return erc20Balance ?? 0

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

File: src/utils/LockOverviewUtils.ts [hunk 5]
```diff
@@ -96,3 +97,32 @@ export const calculateReturn = (
 export const calculateGasBuffer = (gasFee: number) => {
   return gasFee + gasFee * 0.1
 }
+
+export const convertBigIntToNumber = (
```
Comment (kesar) [inline src/utils/LockOverviewUtils.ts:101]: there is already a BigNumber.toNumber( ) function and this one converts to Float (which might apply an unexpected rounding)

File: src/components/lock/StakeIQ.tsx (hunk 3 above)
Comment (Emmanueldmlr) [reply src/components/lock/StakeIQ.tsx:106]: because the value can take be in a different format (Result format -when the user clicks the max button to max amount-, string format when the user inputs the lock value in the input field)...we have this complication because we don't want to convert the contract result directly to a number format. But I understand what you are saying.

File: src/utils/LockOverviewUtils.ts (hunk 5 above)
Comment (Emmanueldmlr) [reply src/utils/LockOverviewUtils.ts:101]: https://docs.ethers.io/v5/api/utils/bignumber/#:~:text=which%20support%20them.-,BigNumber,-.toNumber(

I was getting this error when I tried it.

File: src/hooks/useLock.ts (hunk 1 above)
Comment (kesar) [reply src/hooks/useLock.ts:53]: oh yeah, good point 👍🏻 . but the problem is that I think we talked to use bigInts, and here we are sending number. ideally it should expect bigInt and use formatEther to display proper value

File: src/utils/LockOverviewUtils.ts (hunk 5 above)
Comment (kesar) [reply src/utils/LockOverviewUtils.ts:101]: if you get a min/max error it means that code is wrong somewhere, hiding that error using another function that does parsing and rounding its not a proper solution

File: src/components/lock/StakeIQ.tsx (hunk 3 above)
Comment (kesar) [reply src/components/lock/StakeIQ.tsx:106]: when user clicks max button it should take bigInt value that should be stored in the state
when user input, string should be converted to bigInt and sent to that function

File: src/utils/LockOverviewUtils.ts (hunk 5 above)
Comment (Emmanueldmlr) [reply src/utils/LockOverviewUtils.ts:101]: I understand better now. Let me implement that flow now.

**Comment by Emmanueldmlr** [comment]
@kesar...You can take a look at the Implementation now...(I don cook finish 😂 😂 😂)

**Review by JustinCast — APPROVED** [review]
It worked for me 🚀https://goerli.etherscan.io/tx/0xf81337df9a128bdb9fcc57f47e4f5e7d307892883107a91ad52505920fd6b2c7

**Comment by kesar** [comment]
looks good. going to merge it since i dont think it can be worse than what we have now. Lets still test it heavily on goerli @Emmanueldmlr . I will test it too in mainnet

**Comment by Emmanueldmlr** [comment]
Nice, I will test for different edge cases.
